src/main/java/net/imagej/ops/features/hog/HistogramOfOrientedGradients2D.java

```diff
@@ -0,0 +1,245 @@
+ * 
+ * @author Simon Schmid (University of Konstanz)
+ */
+@Plugin(type = Ops.HoG.HistogramOfOrientedGradients.class)
+public class HistogramOfOrientedGradients2D<T extends RealType<T>>
+		extends AbstractUnaryHybridCF<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>>
+		implements Ops.HoG.HistogramOfOrientedGradients, Contingent {
+
+	@Parameter(required = false)
+	private int numOrientations;
+
+	@Parameter(required = false)
+	private int spanOfNeighborhood;
+
+	@SuppressWarnings("rawtypes")
+	private UnaryFunctionOp<FinalInterval, RandomAccessibleInterval> createOp;
+
+	private ExecutorService es;
+
+	@Parameter
+	private ThreadService ts;
+
+	@Override
+	public void initialize() {
+		es = ts.getExecutorService();
+
+		if (numOrientations == 0)
+			numOrientations = 9;
+		if (spanOfNeighborhood == 0)
+			spanOfNeighborhood = 2;
+
+		createOp = Functions.unary(ops(), CreateImgFromDimsAndType.class, RandomAccessibleInterval.class,
+				new FinalInterval(in().dimension(0), in().dimension(1), numOrientations), new FloatType());
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public RandomAccessibleInterval<T> createOutput(RandomAccessibleInterval<T> in) {
+		return createOp.compute1(new FinalInterval(in().dimension(0), in().dimension(1), numOrientations));
+	}
+
+	@Override
+	public boolean conforms() {
+		return in().numDimensions() == 2;
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public void compute1(RandomAccessibleInterval<T> in, RandomAccessibleInterval<T> out) {
+		RandomAccessible<FloatType> convertedIn = Converters.convert(Views.extendMirrorDouble(in),
+				new Converter<T, FloatType>() {
+					@Override
+					public void convert(final T arg0, final FloatType arg1) {
+						arg1.setReal(arg0.getRealFloat());
+					}
+				}, new FloatType());
+
+		// compute partial derivative for each dimension
+		RandomAccessibleInterval<FloatType> derivative0 = ops().create().img(in, new FloatType());
+		RandomAccessibleInterval<FloatType> derivative1 = ops().create().img(in, new FloatType());
+		PartialDerivative.gradientCentralDifference(convertedIn, derivative0, 0);
+		PartialDerivative.gradientCentralDifference(convertedIn, derivative1, 1);
+
+		// store them in a CompositeView
+		CompositeIntervalView<FloatType, RealComposite<FloatType>> derivatives = Views
+				.collapseReal(Views.stack(derivative0, derivative1));
+
+		// compute angles and magnitudes
+		RandomAccessibleInterval<FloatType> angles = ops().create().img(in, new FloatType());
+		RandomAccessibleInterval<FloatType> magnitudes = ops().create().img(in, new FloatType());
+		CompositeView<FloatType, RealComposite<FloatType>>.CompositeRandomAccess raDerivatives = derivatives
+				.randomAccess();
+		RandomAccess<FloatType> raMagnitudes = magnitudes.randomAccess();
+		RandomAccess<FloatType> raAngles = angles.randomAccess();
+		for (int i = 0; i < in.dimension(0); i++) {
+			for (int j = 0; j < in.dimension(1); j++) {
+				final long[] pos = new long[] { i, j };
```

Comment:
reuse array or use 3 `Cursor's if `angles`, `magnitudes` and `derivatives` have the same `IterationOrder`.
Or: Just set the position of one `RA` and set the others on the position of this one. `RA`s are `Localizable`.